packages/block-editor/src/components/inserter/index.js

```diff
@@ -274,7 +284,12 @@ export default compose( [
 
 				const { insertBlock } = dispatch( blockEditorStore );
 
-				const blockToInsert = createBlock( allowedBlockType.name );
+				const blockToInsert = directInsertBlock?.length
+					? createBlock(
+							directInsertBlock[ 0 ],
+							directInsertBlock[ 1 ]
+					  )
+					: createBlock( allowedBlockType.name );
```

Comment:
I think the default block can work as an array or object. It may help to give a few more hints on the data structure with something like:

```jsx
const [ name, attributes, innerBlocks ] = defaultBlock;
createBlock( name, attributes, innerBlocks );
```

Reply:
Can you think of a scenario where we'd need defaultBlock to be an object? In the existing use cases (both this and your other PR) we're using it to store values to pass to `createBlock` so it makes more sense as an array. 

In this PR, it's defined as an array in the jsdocs for the parameters passed to `useNestedSettingsUpdate` and the return value of `__experimentalGetDirectInsertBlock`. We'd need to change those if we wanted to support an object too.

Reply:
For me, the only reason to make it a block would be an API like `const DEFAULT_BLOCK = createBlock( 'core/navigation-link' )`. One one hand, it makes the logic apparent. On the other, it carries uuid and a few other things we don't need here. Maybe if that was a function? `const DEFAULT_BLOCK_CREATOR = () => createBlock( 'core/navigation-link' )`

Reply:
🤔 Good point, I suppose folks will need to use createBlock for complex innerBlocks and we'll want to avoid keeping around a static uuid for blocks. A function would make sense in that case, though we might want to pick a friendlier name. (I find that some folks get confused by the `creator` name).  Maybe `INSERT_DEFAULT_BLOCK` or something similar?

Reply:
Hmm, I'm not so sure about this. When allowedBlocks is in use, the block is created directly in the inserter. I'd prefer to emulate that logic: fetching a block type, and creating the block from that type in the inserter. Default block and allowed block shouldn't be too different, because they essentially serve the same purpose. The main difference is that only one default block may exist even if multiple blocks are allowed.

> I suppose folks will need to use createBlock for complex innerBlocks and we'll want to avoid keeping around a static uuid for blocks

I'm not sure I follow. We shouldn't have any reason to pass innerBlocks to the inserter, because it should always insert a brand new block.

Reply:
> I think @gwwar referred to pre-populating that block with inner blocks. For example, that could be a pre-populated menu like this:

That's right. A better example might be a container block that makes little sense without children, for example the buttons block.

Since we don't have an explicit case for passing though attributes or innerBlocks yet, I think it's fine deferring needing to pass through extra arguments until we need them. Whichever combination you prefer @tellthemachines

Reply:
```jsx
const blockToInsert = directInsertBlock?.length
					? createBlock( ...directInsertBlock )
					: createBlock( allowedBlockType.name )
```

One thing to note is that if folks pass through innerBlocks in an object, it's possible to have innerBlocks with static uuids. It's an edge case, but two options I can think of:

1. clone the block
```jsx
insertBlock( cloneBlock( blockToInsert ), getInsertionIndex(), rootClientId );
```
2. Avoid passing through innerBlocks if we don't want to handle this case yet 
```jsx
const [ name, attributes ] = directInsertBlock;
createBlock( name, attributes )
```